### PR TITLE
Make scheduler use lastExecutedAt instead of next planned executeAt

### PR DESCRIPTION
**Pull request by evlekht**
- Improves scheduler initial timer code to support canceling by context.
- Modifies scheduled job: instead of next execution time, job now stores last execution time.

## Summary by CodeRabbit

- **New Features**
	- Enhanced job scheduling and execution logic by replacing the `ExecuteAt` field with `LastExecutedAt`, improving tracking of job execution times.
- **Bug Fixes**
	- Adjusted job execution timing calculations to ensure accurate scheduling based on the new `LastExecutedAt` field.
- **Tests**
	- Updated test cases to reflect changes in job attributes and ensure consistency in job execution timing.
- **Chores**
	- Refined logging capabilities during database migrations for better tracking of migration events.

### CODE REVIEW

**Review by havan — DISMISSED** [review]

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -164,22 +171,16 @@ func (s *scheduler) Schedule(ctx context.Context, period time.Duration, jobName
...
-		if executeAt.Before(job.ExecuteAt) {
-			job.ExecuteAt = executeAt
-		}
-	} else {
-		job = &Job{
-			Name:      jobName,
-			ExecuteAt: executeAt,
-			Period:    period,
-		}
+		lastExecutedAt = job.LastExecutedAt
```
Comment (VjeraTurk) [inline pkg/scheduler/scheduler.go:176]: @evlekht please update comment on line 159

**Review by VjeraTurk — APPROVED** [review]

File: migrations/scheduler/2_last_executed_at.up.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE jobs
+DROP execute_at;
```
[the remark this answers is not in the document]
Comment (evlekht) [reply migrations/scheduler/2_last_executed_at.up.sql:2]: I just tested it with sqlite driver version that is used in this app - its working, migration is performed correctly.

File: migrations/scheduler/2_last_executed_at.down.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE jobs
+DROP last_executed_at;
```
[the remark this answers is not in the document]
Comment (evlekht) [reply migrations/scheduler/2_last_executed_at.down.sql:2]: I just tested it with sqlite driver version that is used in this app - its working, migration is performed correctly.

File: pkg/scheduler/scheduler.go [hunk 4]
```diff
@@ -96,37 +96,44 @@ func (s *scheduler) Start(ctx context.Context) error {
...
 		onceDone := make(chan struct{})
 
-		handler := func() {
+		handler := func(tickTime time.Time) {
 			// TODO @evlekht panic handling?
-			if err := s.updateJobExecutionTime(ctx, jobName); err != nil {
+			if err := s.updateJobExecutionTime(ctx, jobName, tickTime); err != nil {
 				s.logger.Errorf("failed to update job execution time: %v", err)
 				return // TODO @evlekht handle error, maybe retry ?
 			}
```
[the remark this answers is not in the document]
Comment (evlekht) [reply pkg/scheduler/scheduler.go:111]: Those TODO comments are left to be resolved in future PRs.

[the remark this answers is not in the document]
Comment (evlekht) [reply pkg/scheduler/scheduler.go:111]: Keep in mind that its only relevant for this PR's TODOs. You should keep asking about newly introduced TODOs in next PRs like you did this time.
And no, we don't need an issue for that.

**Pull request merged**